src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,801 @@
+            this.application = application;
+            this.routeId = routeId;
+            this.initialId = initialId;
+            this.replyId = replyId;
+        }
+
+        public void onApplication(
+            int msgTypeId,
+            DirectBuffer buffer,
+            int index,
+            int length)
+        {
+            switch(msgTypeId)
+            {
+                case BeginFW.TYPE_ID:
+                    final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+                    onApplicationBegin(begin);
+                    break;
+                case DataFW.TYPE_ID:
+                    final DataFW data = dataRO.wrap(buffer, index, index + length);
+                    onApplicationData(data);
+                    break;
+                case EndFW.TYPE_ID:
+                    final EndFW end = endRO.wrap(buffer, index, index + length);
+                    onApplicationEnd(end);
+                    break;
+                case WindowFW.TYPE_ID:
+                    final WindowFW window = windowRO.wrap(buffer, index, index + length);
+                    onApplicationWindow(window);
+                    break;
+                case ResetFW.TYPE_ID:
+                    final ResetFW reset = resetRO.wrap(buffer, index, index + length);
+                    onApplicationReset(reset);
+                    break;
+            }
+        }
+
+        private void onApplicationWindow(
+            WindowFW window)
+        {
+            //TODO
+        }
+
+        private void onApplicationReset(
+            ResetFW reset)
+        {
+            //TODO
+        }
+
+        private void onApplicationBegin(
+            BeginFW begin)
+        {
+
+            OctetsFW extension = begin.extension();
+            SocksBeginExFW socksBeginEx = extension.get(socksBeginExRO::wrap);
+
+            String address = socksBeginEx.address().asString();
+            int port = socksBeginEx.port();
+
+            this.receiver.doSocksCommandReply(address, port);
+
+        }
+
+        private void onApplicationData(
+            DataFW data)
+        {
+            final OctetsFW payload = data.payload();
+            //TODO
+        }
+
+        private void onApplicationEnd(
+            EndFW end)
+        {
+            final long traceId = end.trace();
+            doApplicationEnd(traceId);
+        }
+
+        private void doApplicationBegin(
+            MessageConsumer target,
+            long streamId,
+            long traceId,
+            String address,
+            int port)
+        {
+
+            SocksBeginExFW socksBeginEx = socksBeginExRW.wrap(extBuffer, 0, extBuffer.capacity())
+                .typeId(socksTypeId)
+                .address(address)
+                .port(port)
+                .build();
+
+            final BeginFW begin = beginRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                                         .routeId(this.routeId)
```

Comment:
Please remove redundant `this.` prefix.
Also, why is `streamId` passed into this method, is it different from `initialId`? If not, please use `initialId` instead and remove `streamId` parameter from the method signature.